bentoml/service.py

```diff
@@ -199,16 +199,16 @@ def _validation(inf_tasks):
                     task.discard(http_status=400, err_msg=str(e))
 
         # extract args
-        user_args = self.input_adapter.extract_user_func_args(_validation(inf_tasks))
+        user_args = self.input_adapter.extract_user_func_args(valid_tasks(inf_tasks))
         contexts = tuple(t.context for t in inf_tasks if not t.is_discarded)
 
         # call user function
         user_return = self.user_func(*user_args, contexts=contexts)
 
         if (
-            isinstance(user_return, (tuple, list))
+            isinstance(user_return, (Sequence, Iterable))
```

Comment:
Generator is also a Iterable, but shouldn't be allowed. Otherwise the type checking above would consume some elements.